src/components/Integrations/Shopify/Shopify.js

```diff
@@ -61,35 +62,66 @@ const Shopify = ({ intl, dependencies: { shopifyClient } }) => {
       </tbody>
     </table>
   );
+
+  const shopifyHeader = (
+    <>
+      <div className="block">
+        <h2>{_('shopify.header_title')}</h2>
+        <p>
+          <FormattedHTMLMessage id="shopify.header_subtitle" />
+        </p>
+      </div>
+      <hr />
+    </>
+  );
+
+  function useInterval(callback, delay) {
+    const savedCallback = useRef();
+
+    useEffect(() => {
+      savedCallback.current = callback;
+    }, [callback]);
+
+    useEffect(() => {
+      function tick() {
+        savedCallback.current();
+      }
+      if (delay !== null) {
+        let id = setInterval(tick, delay);
+        return () => clearInterval(id);
+      }
+    }, [delay]);
+  }
+
   useEffect(() => {
     const getData = async () => {
       const result = await shopifyClient.getShopifyData();
       if (!result.success) {
         setError(<FormattedHTMLMessage id="validation_messages.error_unexpected_HTML" />);
       } else if (result.value.length) {
         setIsConnected(true);
+        setDoPoll(0);
         setShops(result.value);
         setError(null);
       } else {
         setIsConnected(false);
         setError(null);
         setShops(result.value);
       }
-      setIsLoading(false);
+      if (!doPoll) {
+        setIsLoading(false);
+      }
     };
     getData();
-  }, [shopifyClient]);
+  }, [shopifyClient, doPoll]);
 
-  const shopifyHeader = (
-    <>
-      <div className="block">
-        <h2>{_('shopify.header_title')}</h2>
-        <p>
-          <FormattedHTMLMessage id="shopify.header_subtitle" />
-        </p>
-      </div>
-      <hr />
-    </>
+  useInterval(
+    () => {
+      if (isLoading) {
+        setDoPoll((doPoll) => doPoll + 1);
```

Comment:
What is the meaning of this counter?

Reply:
It's a counter to trigger the effect of getShops data. See useEffect is dependant on this counter.

Reply:
Mmm, it is difficult to understand. 

What do you think about doing the request in the callback?

Reply:
Seems a good idea, I'll see to it.

Reply:
It worked wonders! Thanks @andresmoschini !